tasks/notify.py

```diff
@@ -130,6 +138,30 @@ async def run_async_within_lock(
             Commit.repoid == repoid, Commit.commitid == commitid
         )
         commit = commits_query.first()
+
+        # check if there were any test failures
+        latest_test_instances = (
+            db_session.query(TestInstance)
+            .join(Upload)
+            .join(CommitReport)
+            .filter(
+                CommitReport.commit_id == commit.id_,
+            )
+            .order_by(TestInstance.test_id)
```

Comment:
any particular reason?

Reply:
it's how the sql works, because I'm using `distinct on` on the `test_id` to select only once test instance for each `test_id`.  `distinct on` requires the left most ordered field to be the same as the field we are `distinct on`: https://www.postgresql.org/docs/9.0/sql-select.html#SQL-DISTINCT